op-node/rollup/derive/engine_queue.go

```diff
@@ -683,6 +706,51 @@ func (eq *EngineQueue) forceNextSafeAttributes(ctx context.Context) error {
 	return nil
 }
 
+// tryBackupUnsafeReorg tries to reorg unsafe head to backupUnsafeHead.
+func (eq *EngineQueue) tryBackupUnsafeReorg(ctx context.Context) error {
+	defer func() {
+		// Reset backupUnsafeHead to avoid multiple usage.
+		eq.backupUnsafeHead = eth.L2BlockRef{}
+		// Only try once because execution engine may forgot backupUnsafeHead
+		// or backupUnsafeHead is not part of the chain.
+		eq.checkBackupUnsafeReorg = false
+	}()
+	if eq.backupUnsafeHead == (eth.L2BlockRef{}) { // sanity check backupUnsafeHead is there
+		return nil
+	}
+	if eq.backupUnsafeHead.Number <= eq.unsafeHead.Number {
+		eq.log.Info("current safe head is already the longest unsafe chain", "backupUnsafe", eq.backupUnsafeHead.ID(), "unsafe", eq.unsafeHead.ID())
+		return nil
+	}
+	// Reorg unsafe chain. Safe/Finalized chain will not be updated.
+	eq.log.Warn("trying to restore unsafe head", "backupUnsafe", eq.backupUnsafeHead.ID(), "unsafe", eq.unsafeHead.ID())
+	fc := eth.ForkchoiceState{
+		HeadBlockHash:      eq.backupUnsafeHead.Hash,
+		SafeBlockHash:      eq.safeHead.Hash,
+		FinalizedBlockHash: eq.finalized.Hash,
+	}
+	fcRes, err := eq.engine.ForkchoiceUpdate(ctx, &fc, nil)
+	if err != nil {
+		return err
+	}
+	if fcRes.PayloadStatus.Status == eth.ExecutionValid {
+		// Execution engine accepted the reorg.
+		eq.log.Info("successfully reorged unsafe head", "unsafe", eq.backupUnsafeHead.ID())
+		eq.unsafeHead = eq.backupUnsafeHead
+		eq.engineSyncTarget = eq.backupUnsafeHead
+		eq.metrics.RecordL2Ref("l2_unsafe", eq.unsafeHead)
+		eq.metrics.RecordL2Ref("l2_engineSyncTarget", eq.engineSyncTarget)
+		eq.logSyncProgress("unsafe head reorg using backup")
+	} else {
+		// Execution engine could not reorg back to previous unsafe head.
+		// If ExecutePayloadStatus is
+		//  - INVALID, backupUnsafeHead is forgot, or reorg failed.
+		//  - SYNCING, backupUnsafeHead was seen before, but not part of the chain.
+		return eth.ForkchoiceUpdateErr(fcRes.PayloadStatus)
```

Comment:
Can you wrap the error to give explicit context for the error?

Reply:
Fixed at https://github.com/testinprod-io/optimism/pull/44/commits/58db1dd38b4dc50405c11973b5df379b38f65883